froide/foirequest/forms/request.py

```diff
@@ -449,3 +450,43 @@ def save(self) -> List[str]:
             m for m in trigger.apply_actions(self.foirequest, self.request) if m
         ]
         return messages
+
+
+class RedactDescriptionForm(forms.Form):
+    description = forms.CharField(required=False)
+    description_length = forms.IntegerField()
+
+    def clean_description(self):
+        val = self.cleaned_data["description"]
+        if not val:
+            return []
+        try:
+            val = [int(x) for x in val.split(",")]
+        except ValueError:
+            raise forms.ValidationError("Bad value")
+        return val
+
+    def save(self, request):
+        redacted_description = apply_user_redaction(
+            request.description,
+            self.cleaned_data["description"],
+            self.cleaned_data["description_length"],
+        )
+        first_message = request.first_outgoing_message
+        if request.description_redacted in first_message.plaintext_redacted:
+            first_message.plaintext_redacted = first_message.plaintext_redacted.replace(
+                request.description_redacted, redacted_description
+            )
+            first_message.save()
+        else:
+            messages.add_message(
+                request,
```

Comment:
This needs to be a `HttpRequest` not a `FoiRequest`. Possibly return a value and handle 'flash messages' in the view instead?